source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -289,6 +291,36 @@ void Filter::onDestroy() {
   }
 }
 
+bool Filter::checkAndApplyHeaderMutation(
+    Filters::Common::MutationRules::CheckOperation operation, absl::string_view key,
+    absl::string_view value,
+    const std::function<void(Http::LowerCaseString, absl::string_view)> mutation_func) {
+  if (config_->validateMutations() && (!Http::HeaderUtility::headerNameIsValid(key) ||
+                                       !Http::HeaderUtility::headerValueIsValid(value))) {
+    ENVOY_STREAM_LOG(trace, "Rejecting invalid header '{}':'{}'.", *decoder_callbacks_, key, value);
+    rejectResponse();
+    return false;
+  }
+  // Check header mutation is valid according to configured decoder mutation rules.
+  Http::LowerCaseString lowercase_key(key);
+  switch (config_->checkDecoderHeaderMutation(operation, lowercase_key, value)) {
+  case CheckResult::OK:
+    ENVOY_STREAM_LOG(trace, "'{}':'{}'", *decoder_callbacks_, key, value);
+    if (mutation_func != nullptr) {
+      mutation_func(lowercase_key, value);
+    }
```

Comment:
The `lowercase_key` will be copied when we run the lambda. And the lambda doesn't simplify the code.

So, as my previous suggestion, I think we should execute the mutation out of this function.